source/kernel/src/registry/mod.rs

```diff
@@ -81,6 +83,22 @@ pub struct UserResponse<U, E> {
     uuid: Uuid,
     nonce: u32,
     reply: Result<U, E>,
+    //
+    // KEEP IN SYNC WITH POSTCARD_MAX_SIZE BELOW!
+    //
+}
+
+// UserResponse
+
+impl<U: MaxSize, E: MaxSize> MaxSize for UserResponse<U, E> {
+    //
+    // KEEP IN SYNC WITH STRUCT DEFINITION ABOVE!
+    //
+    const POSTCARD_MAX_SIZE: usize = {
+        <[u8; 16] as MaxSize>::POSTCARD_MAX_SIZE
+            + <u32 as MaxSize>::POSTCARD_MAX_SIZE
+            + <Result<U, E> as MaxSize>::POSTCARD_MAX_SIZE
```

Comment:
would be really cool if this was derivable (or even just declarative-macro-able)...i might try and hack that together for a subsequent branch...

Reply:
So, the macro is derivable, but the problem is that `UUID` is an extern type, so I'd have to do some newtype shenanigans on it to impl the MaxSize trait for it. I was being lazy, but that might be a better idea.

Reply:
ah, i didn't think of that part. fair point.